components/Common/DarkModeToggle/__tests__/index.test.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { LocaleProvider } from '../../../../providers/localeProvider';
+import DarkModeToggle from '../index';
+import { AppProps } from '../../../../types';
+
+const i18nData = {
```

Comment:
Reminder: Remove the `i18nData`. You don't need it, neither you need the `LocaleProvider`, you can simply just import `IntlProvider` (so that it becomes agnostic to our internal logic)